### ISSUE TITLE
DATABASE: Migrate tables in ref_master.accdb to biosum_ref.db

### ISSUE THREAD

**Issue by lbross** [USFS-PNW/Fia-Biosum-Manager#376]
Do the tables we keep using need to be edited for specific projects? If so, we should move them instead to a ref_master.db or master.db instead of the centralized biosum_ref.db. I'll also indicate which tables we may not need to migrate.
Continue using these tables:
harvest_methods --> biosum_ref.db
owner_groups --> biosum_ref.db
site_index_equations (this one is already slated to move to biosum_ref.db)
tree_species (this one sometimes needs to be edited so it should stay at the project-level. --> master.db?)

These tables may no longer be needed:
FVS_EasternTreeSpeciesTranslator
FVS_WesternTreeSpeciesTranslator
fvs_rx_category
fvs_rx_subcategory (I think this and the table above were used to source values for the rx_category and rx_subcategory tables that we are removing as part of the fvs_master.mdb migration)
TreeMacroPlotBreakPointDia (I'm struggling to be sure what this was/is used for. The macro stuff is used for substituting values for a fixed value but we're not using it many places. My version of this table only includes values for 3 states which doesn't support nationwide availability)

**Comment by jsfried** [comment]
In agreement on most of this. TreeMacroPlotBreakPointDia is actually in the FIADB data and is specific to each plot in the PNW states (macroplot is not used in other states). It might be needed (or might have been needed) in calculating tree expansions as it determines which trees are sampled on the macro plot-- i.e., either trees larger than 24 inches (in California, and on the east side in OR and WA I think), or larger than 30 inches (west side OR and WA but maybe also in the NW Forest Plan area parts of CA-- can check the value in PLOT to be sure). I guess I'd need to see exactly how it is used in BioSum to know whether we need to keep it.

**Comment by lbross** [comment]
We will retain TreeMacroPlotBreakPointDia but move it to biosum_ref.db. This table is used during the condprop calculations for OR, WA, CA. It's also available as a macro variable in the Optimizer condition and plot filters. We may postpone migrating these tables until the plot input migration because most are primarily used there.

**Comment by lbross** [comment]
Notes to tester:

1. Check for the new biosum_ref.db in the user AppData folder
2. The tables that are being used from here by v5.11.2 are site_index_equations, REF_FOREST_TYPE_GROUP, REF_FOREST_TYPE, REV_VERSION and harvest_methods. The remaining tables are placeholders be used in future versions.
3. site_index_equations, REF_FOREST_TYPE_GROUP, REF_FOREST_TYPE are used by FVSIn and can be validated while testing #356 
4. harvest_methods is used by the Rx and Processor screens
5. Check the project and processor data sources to make sure links to these tables aren't broken for upgraded projects

**Comment by Cat-Ducat-USDA** [comment]
I have validated that the new biosum_ref.db is showing up in the Appdata folder, populated by the correct tables; however, it seems that owner_groups is still being sourced from ref_master.accdb and FIA REF tables are being sourced from biosum_ref.accdb. Can't find the site_index_equations table/source under any data sources tabs. Is it called something else there? harvest_methods IS being pulled from the new biosum_ref.db, but I think that's it. No links are broken, though.

**Comment by jsfried** [comment]
@Cat-Ducat-USDA 

So my presumption is that it is intended that both biosum_ref databases are in use during the transition, with some tables accessed from one and some from the other. 

Lesley's notes to tester indicate that 1) site_index_equations, 2) REF_FOREST_TYPE_GROUP, 3) REF_FOREST_TYPE, 4) REV_VERSION and 5) harvest_methods are the 5 to pay attention to (she does say the others are "place holders") and, beyond verifying that they are populated, through validation calls for investigating parts of BioSum that pull from these tables to ensure that they are, in fact, pulling from these tables. 

It helps to know what these tables are used for so that you can know where to look. #1 is used to calculate site index for conditions where it is not populated in the FIADB.FVS_STANDINIT_COND (which will be null in cases where FVSIn.FVS_STANDINIT_COND is populated and not null by virtue of being populated through BioSum's calculations). If you can find cases of stands with this discrepancy between the two tables (null vs populated) that verifies that BioSum is populating by making reference to this table. Verification that the right SI equation (supplied by this lookup table) is being used might be outside the scope of what we as analysts without access to the code can easily verify-- hopefully Drue or Lesley have been able to spotcheck that). 

#2 and #3's use could be checked, I think, by (after making a backup of the biosum_ref.accdb database) deleting those tables from the accdb and making sure that when you do a fresh FVSIn creation with Down Wood included, you end up with an FVS_STANDINIT_COND table with values for duff and litter and fine wood fuel loadings (since these tables provide inputs to the calculation of those fields). 

#4 is internal to BioSum and as long as it is different than in the ACCDB, that's likely fine. 

#5 supplies info to the processor screen where you can select a harvest system (to override Rx specific selections) and see its description in a text box-- I would suggest changing one word in the description in the table and then make sure it shows up on that screen when you select that harvest system. You could also (before doing these explorations) remove or rename the tables harvest_methods and site_index_equations in the ref_master.mdb in the project\db folder to provide assurance that 5.11.2 is using the new tables, not the old ones. Right off the bat I am noticing some differences for CWI Demo in that the ref_master.accdb version of the site_index_equations has only 31 rows and none that address the EC variant whereas the biosum_ref version has 231 representing all the variants used in the west coast states and inland northwest (but not Alaska)

I recommend following this logic as you complete validation for this issue.

**Comment by lbross** [comment]
Yes. owner_groups and the FIA REF tables are still being addressed in their .accdb locations. I thought we were going to skip v5.11.2 and was trying to get ahead of things adding these tables to the biosum_ref.db.

- site_index_equations: I will let @druepdx respond to the scope of testing. The difference in the row count is because in v5.11.1 we were using the table for 4 variants and in v5.11.2 we are using the table for ALL variants
- ref_version keeps track of the current version(s) of the biosum_ref.accdb and the biosum_ref.db respectively. The values will be different

The rest of @jsfried suggestions are spot on.

**Comment by Cat-Ducat-USDA** [comment]
Got it, thanks for the clarification! I will spend some time with this shortly.

**Comment by jsfried** [comment]
Jeremy will pick up validation on this and is awaiting scope advice from Drue.

**Comment by jsfried** [comment]
1.	Checked item #5 -- that harvest methods now hitting off of the biosum_ref.db in app data.
2.	Fia_Tree_species_Ref in the biosum_ref.accdb in appdata is still required. Apparently not using the same named table in BioSum_Species_Ref.accdb. Plot load error if this table does not exist.
3.	Items #2 and #3 above checked. 
4.	For issue #1 above, we crafted a query linking the STANDINIT_COND tables created by BioSum and the one in FIADB:

SELECT FVS_STANDINIT_COND.STAND_CN, FVS_STANDINIT_COND.SAM_WT, FVS_STANDINIT_COND_FIADB.SAM_WT, cond.condprop, cond.condprop_unadj, FVS_STANDINIT_COND.SITE_INDEX, FVS_STANDINIT_COND_FIADB.SITE_INDEX, FVS_STANDINIT_COND.SITE_SPECIES, FVS_STANDINIT_COND_FIADB.SITE_SPECIES, FVS_STANDINIT_COND.SITE_INDEX_BASE_AG, FVS_STANDINIT_COND_FIADB.SITE_INDEX_BASE_AG
FROM (FVS_STANDINIT_COND INNER JOIN FVS_STANDINIT_COND_FIADB ON FVS_STANDINIT_COND.STAND_CN = FVS_STANDINIT_COND_FIADB.STAND_CN) INNER JOIN cond ON FVS_STANDINIT_COND.STAND_ID = cond.biosum_cond_id;
Which returned these rows (below).

Things to notice:
1)	There are three cases where BioSum calculated a site index where NIMS/FIADB compilation did not—2 were Mtn Hemlock and one was Douglas-fir. The Douglas-fir site index was calculated from DF site trees in a low productivity red alder stand (generated only ~1000 ft3 of wood in 85 years so much less than 20 cu ft per ac per yr threshold for timberland [sites higher than 7]). The hemlock site trees were used in hemlock stands which were also very low site quality (7). Curiously, the base age values for the SI equations provided via BioSum for these stands are not entered, but should be.

2)	Information only (mainly for Cat): SAM_WT (sample weight) in the first two columns differs between BioSum and FIADB sourced  data. Not sure how FIADB is calculating this. We put the value of cond.acres in the BioSum sourced version, which is derived from POP tables, CONDPRO_UNADJ and how many too-small conditions are dropped in the FIADB load op (since the area represented by those conditions is reallocated to those conditions maintained in BioSum). Upshot- don’t expect these SAM_WT to be the same—they won’t be. Included condprop (adjusted by implication, when not named that way) and condprop_unadj for illustrative purposes—we almost always want unadj for our calcs as adj is tied to the stratification and percent of plot area across the stratum that is nonsampled.  

![Image](https://github.com/user-attachments/assets/e18cddcb-046d-4bd6-9457-db7b836e47b6)

**Comment by lbross** [comment]
owner_groups is still sourced from ref_master.mdb, REF tables are still sourced from biosum_ref.accdb
site_index_equations are coming from biosum_ref.db
Fia_Tree_species_Ref in the biosum_ref.accdb in appdata is still required. I'm not sure of the reference to BioSum_Species_Ref.accdb in @jsfried  comment above. This is not a database I'm familiar with.

**Comment by jsfried** [comment]
My bad then-- must be a database that I generated from the table in another database (and am storing in Appdata)
